client/src/components/Subscriber/Followers/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { Waypoint } from 'react-waypoint';
+import { Button, Spinner } from 'reactstrap';
+
+export interface IFollowersProps {
+    followers: [];
+    urlUsername: string;
+    page: number;
+    loggedId: string;
+    allFollowersLoaded: boolean;
+    loadFollow: boolean;
+    followUser: (body: { _id: string }) => void;
+    unfollowUser: (body: { _id: string }) => void;
+    getFollowers: (loggedId: string, urlUsername: string) => void;
+    setNextPage: (page: number) => void;
+    getMoreFollowers: (loggedId: string, urlUsername: string, page: number) => void;
+    changeUserFollowing: (_id: string) => void;
+}
+
+export class Followers extends React.Component<IFollowersProps> {
+    public componentDidUpdate(prevProps: any): void {
+        if (prevProps.loggedId !== this.props.loggedId) {
+            this.props.getFollowers(this.props.loggedId, this.props.urlUsername);
+        }
+    }
+
+    public getMoreFollowers = (): void => {
+        if (!this.props.allFollowersLoaded && this.props.loggedId) {
+            this.props.setNextPage(this.props.page);
+            this.props.getMoreFollowers(this.props.loggedId, this.props.urlUsername, this.props.page);
+        }
+    }
+
+    public dynamicButton = (_id: string, alreadyFollow: boolean): any => {
+        const followSubscriber = (): void => {
+            const body = {_id};
+            this.props.followUser(body);
+            this.props.changeUserFollowing(_id);
+        };
+
+        const unfollowSubscriber = (): void => {
```

Comment:
Move  outside the method dynamicButton

Reply:
I can't move outside unfollowSubscriber() function  because it needs  _follower._id_  which it gets through
dynamicButton() function.

Reply:
you can use arguments of function

Reply:
fixed